be/src/util/frame_of_reference_coding.cpp

```diff
@@ -109,13 +131,20 @@ void ForEncoder<T>::bit_packing_one_frame_value(const T* input) {
     T max = input[0];
     bool is_ascending = true;
     uint8_t bit_width = 0;
+    T half_max_delta = numeric_limits_max() >> 1;
+    bool save_original_value = false;
 
+    // 1. make sure order_flag, save_original_value, and find max&min.
     for (uint8_t i = 1; i < _buffered_values_num; ++i) {
         if (is_ascending) {
             if (input[i] < input[i - 1]) {
                 is_ascending = false;
             } else {
-                bit_width = std::max(bit_width, bits(input[i]- input[i - 1])) ;
+                if ((input[i] >> 1) - (input[i - 1] >> 1) > half_max_delta) { // overflow
+                    save_original_value = true;
```

Comment:
1. seems can break this loop?
2. if don't break this loop, this can be done after loop with min and max

Reply:
answer for q1: 
In my opinion, `save_original_value` and `is_ascending` are two independent variables even though the difference in is_ascending only affects the encoding format now. If break the loop when delta value > half_max_delta, we can not known the exact order.
answer for q2:
We can use subtraction result of `(max >> 1) - (min >> 1)` to determine whether to save original value only when the order id **not** ascending. When the order is ascending, condition `(max >> 1) - (min >> 1) > half_max_delta` is insufficient to save original value.